src/ModelContextProtocol.Core/AIContentExtensions.cs

```diff
@@ -380,21 +380,21 @@ public static ContentBlock ToContentBlock(this AIContent content)
 
             DataContent dataContent when dataContent.HasTopLevelMediaType("image") => new ImageContentBlock
             {
-                Data = dataContent.Base64Data.ToString(),
+                Data = System.Text.Encoding.UTF8.GetBytes(dataContent.Base64Data.ToString()),
                 MimeType = dataContent.MediaType,
             },
 
             DataContent dataContent when dataContent.HasTopLevelMediaType("audio") => new AudioContentBlock
             {
-                Data = dataContent.Base64Data.ToString(),
+                Data = System.Text.Encoding.UTF8.GetBytes(dataContent.Base64Data.ToString()),
                 MimeType = dataContent.MediaType,
             },
 
             DataContent dataContent => new EmbeddedResourceBlock
             {
                 Resource = new BlobResourceContents
                 {
-                    Blob = dataContent.Base64Data.ToString(),
+                    Blob = System.Text.Encoding.UTF8.GetBytes(dataContent.Base64Data.ToString()),
```

Comment:
I think this should just be `Base64Data` and not go to string and back to bytes.